### ISSUE TITLE
Data will lose if remove a layout

### ISSUE THREAD

**Issue by zareismail** [whitecube/nova-flexible-content#154]
Hi. 
thanks for your awesome work.

There is a bug in your package. follow this scenario:

at first, we have these fields: 

```  
    /**
     * Get the fields displayed by the resource.
     *
     * @param  \Illuminate\Http\Request  $request
     * @return array
     */
    public function fields(Request $request)
    {
           Flexible::make("config")->addLayout("page", "page", [
                Text::make("Ali")
            ])->addLayout("post", "post", [
                Text::make("Ali")
            ]),
    }
```
now create one instance of `page` and `post` and save the Resource. then remove the first layout and go back to the edit page. like below:

``` 

    /**
     * Get the fields displayed by the resource.
     *
     * @param  \Illuminate\Http\Request  $request
     * @return array
     */
    public function fields(Request $request)
    {
           Flexible::make("config")->addLayout("post", "post", [
                Text::make("Ali")
            ]),
    }
```
and we lose other data! why?
I resolved it. 
in the `\WhiteCube\NovaFlexibleContent\Value\Resolver` class, after layouts filtering, at line `35`, the collection will lose ` indexes` of the rejected layouts. you will receive an object instead of an array in JSON serialization.

so; you should change that class like this:
```
/**
     * get the field's value
     *
     * @param  mixed  $resource
     * @param  string $attribute
     * @param  Whitecube\NovaFlexibleContent\Layouts\Collection $layouts
     * @return Illuminate\Support\Collection
     */
    public function get($resource, $attribute, $layouts)
    {
        $value = $this->extractValueFromResource($resource, $attribute);

        return collect($value)->map(function($item) use ($layouts) {
            $layout = $layouts->find($item->layout);

            if(!$layout) return;

            return $layout->duplicateAndHydrate($item->key, (array) $item->attributes);
        })->filter()->values();
    }
```

**Comment by zareismail** [comment]
resolved in PR #155

**Comment by Nyratas** [comment]
Thanks for noticing @zareismail !

I just merged it and will publish a new release shortly.

Have a great day.
